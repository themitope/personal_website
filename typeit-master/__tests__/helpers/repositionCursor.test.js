import repositionCursor from "../../src/helpers/repositionCursor";
import getAllTypeableNodes from "../../src/helpers/getAllTypeableNodes";

let element, cursor, allCharacters;

beforeEach(() => {
  setHTML`<span id="el"></span>`;

  element = document.querySelector("#el");

  // These need to be individual text nodes so that
  // `getAllTypeableNodes` can correctly collect individual nodes.
  [1, 2, 3, 4, 5, '<i id="cursor">|</i>'].forEach(thing => {
    element.insertAdjacentHTML("beforeend", thing);
  });

  cursor = document.querySelector("#cursor");
  allCharacters = getAllTypeableNodes(element, cursor, true);
});

test("Does not move cursor when stepsToMove is zero.", () => {
  repositionCursor(element, allCharacters, cursor, 0);

  expect(document.body.innerHTML).toMatchSnapshot();
});

test("Moves cursor three steps back.", () => {
  repositionCursor(element, allCharacters, cursor, 3);

  expect(document.body.innerHTML).toMatchSnapshot();
});

test("Moves cursor three back and two forward.", () => {
  repositionCursor(element, allCharacters, cursor, 3);

  repositionCursor(element, allCharacters, cursor, 1);

  expect(document.body.innerHTML).toMatchSnapshot();
});

test("Stops moving when at beginning of string.", () => {
  repositionCursor(element, allCharacters, cursor, 100);

  expect(document.body.innerHTML).toMatchSnapshot();
});

test("Stops moving when at end of string.", () => {
  repositionCursor(element, allCharacters, cursor, -100);

  expect(document.body.innerHTML).toMatchSnapshot();
});
