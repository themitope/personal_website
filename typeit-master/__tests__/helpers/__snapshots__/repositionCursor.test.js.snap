// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does not move cursor when stepsToMove is zero. 1`] = `"<span id=\\"el\\">12345<i id=\\"cursor\\">|</i></span>"`;

exports[`Moves cursor three back and two forward. 1`] = `"<span id=\\"el\\">1234<i id=\\"cursor\\">|</i>5</span>"`;

exports[`Moves cursor three steps back. 1`] = `"<span id=\\"el\\">12<i id=\\"cursor\\">|</i>345</span>"`;

exports[`Stops moving when at beginning of string. 1`] = `"<span id=\\"el\\"><i id=\\"cursor\\">|</i>12345</span>"`;

exports[`Stops moving when at end of string. 1`] = `"<span id=\\"el\\">12345<i id=\\"cursor\\">|</i></span>"`;
